File: android/app/src/main/kotlin/in/cashify/androidtrc/module/engineer/api/ReplacePartRequest.kt
package `in`.cashify.androidtrc.module.engineer.api

import androidx.annotation.Keep
import com.google.gson.annotations.SerializedName

/**
 * Created by Rishika on 20/10/20.
 */

@Keep
class ReplacePartRequest {
    @SerializedName("pbr")
    var partBarcode:String? = null

    @SerializedName("pdbr")
    var previousDeviceBarCode:String? = null  // device on which to remove

    @SerializedName("ndbr")
    var newDeviceBarCode:String? = null // device on which to apply
}

File: flutter_module/lib/src/modules/engineer/my_devices/wip_devices/view_parts/models/replace_part_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'replace_part_request.g.dart';

@JsonSerializable()
class ReplacePartRequest {
  @JsonKey(name: "pbr")
  String? partBarcode;

  @JsonKey(name: "pdbr")
  String? previousDeviceBarCode; // device on which to remove

  @JsonKey(name: "ndbr")
  String? newDeviceBarCode;

  ReplacePartRequest(this.partBarcode, this.previousDeviceBarCode, this.newDeviceBarCode); // device on which to apply

  static ReplacePartRequest fromJson(Map<String, dynamic> json) => _$ReplacePartRequestFromJson(json);

  Map<String, dynamic> toJson() => _$ReplacePartRequestToJson(this);
}
